src/core/lib/iomgr/tcp_server_posix.c

```diff
@@ -736,18 +745,21 @@ void grpc_tcp_server_shutdown_starting_add(grpc_tcp_server *s,
 
 void grpc_tcp_server_unref(grpc_exec_ctx *exec_ctx, grpc_tcp_server *s) {
   if (gpr_unref(&s->refs)) {
-    /* Complete shutdown_starting work before destroying. */
     grpc_exec_ctx local_exec_ctx = GRPC_EXEC_CTX_INIT;
+    bool finish_ctx = false;
+    /* FIXME: API allows a NULL exec_ctx, although this might cause us to delete
```

Comment:
We did. The (repeatedly justified, I feel) fear was that it would be too magical:
- new surface API's calling into gRPC internals would likely miss it, and weird crashes could result (that wouldn't necessarily be correlated with the code that failed to set the thread local)
- missing nullifying the thread local at the end of a scope would lead to very difficult action-at-a-distance bugs to clean up